### PR TITLE
Fix image classification training scripts and readme

### PR DESCRIPTION
**Pull request by hetong007**

### CODE REVIEW

**Comment by mli** [comment]
Job PR-309-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-309/1/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-309-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-309/2/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-309-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-309/3/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-309-5 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-309/5/index.html 
Code coverage of this PR:  vs. Master:

**Comment by hetong007** [comment]
Adding fix for https://github.com/dmlc/gluon-cv/issues/331

**Comment by mli** [comment]
Job PR-309-8 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-309/8/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-309-9 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-309/9/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-309-12 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-309/12/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/classification/imagenet/train_imagenet.py [hunk 1]
```diff
@@ -322,21 +333,16 @@ def train(ctx):
...
-                    y1 = label_transform(Y, classes, eta)
-                    y2 = label_transform(Y[::-1], classes, eta)
-                    label_mixup.append(lam*y1 + (1-lam)*y2)
+                label = mixup_transform(label, classes, lam, eta)
 
-                data = data_mixup
-                label = label_mixup
             elif opt.label_smoothing:
+                hard_label = label
                 label = smooth(label, classes)
```
Comment (d-li14) [inline scripts/classification/imagenet/train_imagenet.py]: So this `label` is not used now? After adding line 362 the `train_metric` seems to be updated with `hard_label`, irrelevant to the smoothed `label`.

Comment (hetong007) [reply scripts/classification/imagenet/train_imagenet.py]: `label` is used in loss calculation.

**Comment by mli** [comment]
Job PR-309-13 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-309/13/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-309-14 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-309/14/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-309-15 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-309/15/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-309-18 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-309/18/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**